File: score_tmp.js
const fs = require('fs');
const m = JSON.parse(fs.readFileSync('/tmp/markets.json'));
const trend = JSON.parse(fs.readFileSync('/tmp/trending.json'));
const dex = JSON.parse(fs.readFileSync('/tmp/dex.json'));

const trendingSyms = new Set((trend.coins || []).map(c => (c.item.symbol || '').toUpperCase()));
const dexSyms = new Set((dex.pairs || []).map(p => ((p.baseToken || {}).symbol || '').toUpperCase()));

const btc = m.find(x => x.symbol === 'btc');
const eth = m.find(x => x.symbol === 'eth');
const btc7 = btc.price_change_percentage_7d_in_currency;
const eth7 = eth.price_change_percentage_7d_in_currency;
console.log('BTC 24h/7d:', btc.price_change_percentage_24h_in_currency.toFixed(2), btc7.toFixed(2));
console.log('ETH 24h/7d:', eth.price_change_percentage_24h_in_currency.toFixed(2), eth7.toFixed(2));
console.log('trending:', [...trendingSyms].sort().join(' '));
console.log('---');

const dedup = new Set(['TRAC', 'KAIA']);
const rows = [];
for (const x of m) {
  const sym = (x.symbol || '').toUpperCase();
  const mc = x.market_cap || 0;
  const vol = x.total_volume || 0;
  const p24 = x.price_change_percentage_24h_in_currency;
  const p7 = x.price_change_percentage_7d_in_currency;
  if (mc < 20e6) continue;
  if (p24 == null || p7 == null) continue;
  let s = 0;
  if (p24 > 0) s += 1;
  if (p7 > 0) s += 1;
  if (p24 > 5 && p7 > 5) s += 2;
  if (trendingSyms.has(sym)) s += 2;
  const vmc = mc ? vol / mc : 0;
  if (vmc >= 0.20) s += 3;
  else if (vmc >= 0.10) s += 2;
  if (p7 > btc7 && p7 > eth7) s += 2;
  if (dexSyms.has(sym)) s += 1;
  rows.push({ s, sym, name: x.name, price: x.current_price, p24, p7, mc, vol, vmc });
}
rows.sort((a, b) => b.s - a.s || b.vmc - a.vmc);
for (const r of rows.slice(0, 22)) {
  const flag = dedup.has(r.sym) ? ' <DEDUP>' : '';
  console.log(
    `${String(r.s).padStart(2)} ${r.sym.padEnd(9)} ${(r.name || '').slice(0, 20).padEnd(20)} ` +
    `$${String(r.price).padEnd(12)} 24h=${r.p24.toFixed(1).padStart(7)} 7d=${r.p7.toFixed(1).padStart(7)} ` +
    `mc=$${(r.mc / 1e6).toFixed(0).padStart(7)}m vol=$${(r.vol / 1e6).toFixed(0).padStart(7)}m vmc=${r.vmc.toFixed(3)}${flag}`
  );
}

File: score_tmp.py
import json

m = json.load(open('/tmp/markets.json'))
trend = json.load(open('/tmp/trending.json'))
dex = json.load(open('/tmp/dex.json'))

trending_syms = set()
for c in trend.get('coins', []):
    i = c['item']
    trending_syms.add((i.get('symbol') or '').upper())

dex_syms = set()
for p in dex.get('pairs', []) or []:
    bt = p.get('baseToken', {})
    dex_syms.add((bt.get('symbol') or '').upper())

btc = next(x for x in m if x['symbol'] == 'btc')
eth = next(x for x in m if x['symbol'] == 'eth')
btc7 = btc['price_change_percentage_7d_in_currency']
eth7 = eth['price_change_percentage_7d_in_currency']
print('BTC 24h/7d:', round(btc['price_change_percentage_24h_in_currency'],2), round(btc7,2))
print('ETH 24h/7d:', round(eth['price_change_percentage_24h_in_currency'],2), round(eth7,2))
print('trending:', sorted(trending_syms))
print('---')

dedup = {'TRAC', 'KAIA'}
rows = []
for x in m:
    sym = (x['symbol'] or '').upper()
    mc = x.get('market_cap') or 0
    vol = x.get('total_volume') or 0
    p24 = x.get('price_change_percentage_24h_in_currency')
    p7 = x.get('price_change_percentage_7d_in_currency')
    if mc < 20_000_000:
        continue
    if p24 is None or p7 is None:
        continue
    s = 0
    if p24 > 0: s += 1
    if p7 > 0: s += 1
    if p24 > 5 and p7 > 5: s += 2
    if sym in trending_syms: s += 2
    vmc = vol / mc if mc else 0
    if vmc >= 0.20: s += 3
    elif vmc >= 0.10: s += 2
    if p7 > btc7 and p7 > eth7: s += 2
    if sym in dex_syms: s += 1
    rows.append((s, sym, x.get('name'), x.get('current_price'), p24, p7, mc, vol, vmc))

rows.sort(reverse=True)
for r in rows[:20]:
    s, sym, name, price, p24, p7, mc, vol, vmc = r
    flag = ' <DEDUP>' if sym in dedup else ''
    print(f"{s:2d} {sym:8s} {(name or '')[:20]:20s} ${price:<12.6g} 24h={p24:7.1f} 7d={p7:7.1f} mc=${mc/1e6:8.0f}m vol=${vol/1e6:8.0f}m vmc={vmc:.3f}{flag}")
